backend/api/routers/dataset_router.py

```
@@ -241,4 +245,59 @@ async def unmount_dataset(
+        if result.is_success:
+            datasets = []
+            for dataset in result.value:
+                datasets.append({
+                    "name": str(dataset.name),
+                    "mountpoint": dataset.properties.get("mountpoint", "")
+                })
+            return {"datasets": datasets}
+        
+        raise HTTPException(status_code=500, detail=f"Failed to list datasets: {result.error}")
+    except SecurityValidationError as e:
+        raise HTTPException(status_code=422, detail=f"Security validation failed: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)